fitbenchmarking/fitbenchmarking
# Cost function class

#### Description of Work

Fixes #489 

Adds new cost function class and removes mention of `r_norm`


#### Testing Instructions

1. Check class does as expected
2. Check there is no mention of `r_norm`

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
